### PR TITLE
[T3A3][T14-A1] Advay Pal

### PR DESCRIPTION
**Pull request by advaypal**
Ready for review.

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -25,8 +29,33 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        String[] addressPartsList = address.split(", ");
+        this.block = new Block(addressPartsList[0]);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: try removing magic numbers

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -25,8 +29,33 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+        this.street = new Street(addressPartsList[1]);
+        this.unit = new Unit(addressPartsList[2]);
+        this.postalCode = new PostalCode(addressPartsList[3]);
     }
+    
+    public Block getBlock() {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: why would you return the objects here? if there is a need to edit the fields, you can use accessor methods, no?

File: test/expected.txt [hunk 3]
```diff
@@ -52,90 +52,90 @@
 || Parameters: NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...
 || Example: add John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney
 || ===================================================
-|| Enter command: || [Command entered:  add Valid Name 12345 e/valid@email.butNoPhonePrefix a/valid, address]
+|| Enter command: || [Command entered:  add Valid Name 12345 e/valid@email.butNoPhonePrefix a/valid, address, a, b]
```
Comment (okkhoy) [inline test/expected.txt]: good to see that you have updated the user guide, tests and in program help :+1:

**Comment by okkhoy** [comment]
@advaypal overall good work :+1:  keep it up.
there are a few minor issues which I have pointed out in the code.
Please check the comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/data/person/Address.java (hunk 1 above)
Comment (advaypal) [inline src/seedu/addressbook/data/person/Address.java]: Yup will do

File: src/seedu/addressbook/data/person/Address.java (hunk 2 above)
Comment (advaypal) [inline src/seedu/addressbook/data/person/Address.java]: Oh yeah, that's a good point. I'll change that

**Comment by advaypal** [comment]
Do I have to push my changes or should I just close the PR?

**Pull request closed without merging**